### PR TITLE
Cow-people can now graze on bushes

### PR DESCRIPTION
**Pull request by Bartimeus973**
[feature]
## About the PR 
Cow people can now click on the bushes spread around the station on help intent to graze on them, gaining minor food buffs and removing visible bites from the bush.
This alleviates the hunger motive on the rp servers.

## Why's this needed? 
Cow-people need a more balanced diet. Also provides a fun way to fix your hunger motive, get some very minor foodbuffs, and also makes the captain's bonsai an interesting target.

## Changelog

```changelog
(u)Bartimeus973
(+)Cow people can now graze on bushes. Watch out for plastic replicas!
```

### CODE REVIEW

File: code/obj/decorations.dm [hunk 1]
```diff
@@ -177,9 +182,56 @@
...
-		if (destroyed) return ..()
+		if (destroyed && iscow(user) && user.a_intent == INTENT_HELP)
+			boutput(user, "<span class='notice'>You pick at the ruined bush, looking for any leafs to graze on, but cannot find any.</span>")
+			return ..()
+		else if (destroyed)
+			return ..()
 
 		user.lastattacked = src
+		if (iscow(user) && user.a_intent == INTENT_HELP)
+			if (istype(src, /obj/shrub/captainshrub))
```
Comment (ZeWaka) [inline code/obj/decorations.dm:193]: Use inheritance properly

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/decorations.dm (hunk 1 above)
Comment (Bartimeus973) [reply code/obj/decorations.dm]: Done, hopefully didnt mess it up.

File: code/obj/decorations.dm [hunk 2]
```diff
@@ -175,11 +180,45 @@
...
+				user.losebreath+=2
+			else
+				user.changeStatus("food_hp_up", 20 SECONDS)
+				user.visible_message("<span class='notice'>[user] takes a bite out of [src].</span>", "<span class='notice'>You munch on some of [src]'s leaves, like any normal human would.</span>")
+				var/mob/living/carbon/human/H = user
+				H.sims?.affectMotive("Hunger", 10)
+
+			if(src.bites <= 0)
+				destroy()
+			return 0
```
Comment (ZeWaka) [inline code/obj/decorations.dm:220]: this should be in its own proc and then you override that in the child

Comment (Bartimeus973) [reply code/obj/decorations.dm]: Done, thanks for the heads up!

File: code/obj/decorations.dm [hunk 3]
```diff
@@ -263,15 +277,47 @@
...
+
+		if (is_plastic)
+			user.setStatus("weakened", 3 SECONDS)
+			user.visible_message("<span class='notice'>[user] takes a bite out of [src] and chokes on the plastic leaves.</span>", "<span class='alert'>You munch on some of [src]'s leaves, but realise too late it's made of plastic. You start choking!</span>")
+			user.take_oxygen_deprivation(20)
+			user.losebreath+=2
+		else
+			user.changeStatus("food_hp_up", 20 SECONDS)
+			user.visible_message("<span class='notice'>[user] takes a bite out of [src].</span>", "<span class='notice'>You munch on some of [src]'s leaves, like any normal human would.</span>")
+			var/mob/living/carbon/human/H = user
```
Comment (ZeWaka) [inline code/obj/decorations.dm:302]: why is it a human

File: code/obj/decorations.dm [hunk 4]
```diff
@@ -159,11 +159,16 @@
 	var/time_between_uses = 400 // The default time between uses.
 	var/override_default_behaviour = 0 // When this is set to 1, the additional_items list will be used to dispense items.
 	var/list/additional_items = list() // See above.
+	var/bites = 5 // How many bites can cow people take out of it?
```
Comment (ZeWaka) [inline code/obj/decorations.dm:162]:
```suggestion
	var/const/bites = 5 // How many bites can cow people take out of it?
```

File: code/obj/decorations.dm [hunk 5]
```diff
@@ -159,11 +159,16 @@
...
+	var/current_mask = 5 // The mask used to show bite marks
+	var/is_plastic = FALSE // Is the bush actually made out of plastic?
```
Comment (ZeWaka) [inline code/obj/decorations.dm:164]: codedoc comments pls

File: code/obj/decorations.dm (hunk 3 above)
Comment (Bartimeus973) [reply code/obj/decorations.dm:302]: I'm checking for "sims" on the next line, which can only be found on humans. Is there another way i can go about this?

File: code/obj/decorations.dm (hunk 4 above)
Comment (ZeWaka) [reply code/obj/decorations.dm]: oh I'm dumb

File: code/obj/decorations.dm (hunk 3 above)
Comment (ZeWaka) [reply code/obj/decorations.dm:304]: How do you know it's a human you didn't check

Comment (ZeWaka) [reply code/obj/decorations.dm]: moreso the argument of the proc doesn't specify that it only takes humans

Comment (Bartimeus973) [reply code/obj/decorations.dm]: Alright yeah that makes more sense

Comment (ZeWaka) [reply code/obj/decorations.dm:304]: unnecessary now

File: code/obj/decorations.dm [hunk 6]
```diff
@@ -175,11 +183,20 @@
...
 				src.take_damage(45)
+
 	attack_hand(mob/user)
 		if (!user) return
-		if (destroyed) return ..()
+		if (destroyed && iscow(user) && user.a_intent == INTENT_HELP)
+			boutput(user, "<span class='notice'>You pick at the ruined bush, looking for any leafs to graze on, but cannot find any.</span>")
+			return ..()
+		else if (destroyed)
+			return ..()
```
Comment (ZeWaka) [inline code/obj/decorations.dm:193]:
```suggestion
		return ..()
```

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**